File: Unsupervised Learning/Dimensionality Reduction/PCA/coding.R
set.seed(42)

X <- matrix(runif(1000), nrow = 200, ncol = 5)

pca_result <- princomp(X)

print("Explained Variance Ratio:")
print(pca_result$sdev^2 / sum(pca_result$sdev^2))


print("Feature loadings:")
print(pca_result$loading)

X_pca <- predict(pca_result)

File: Unsupervised Learning/Dimensionality Reduction/PCA/coding.py
from sklearn.decomposition import PCA
import numpy as np

np.random.seed(42)

# Generate random data
X = np.random.rand(200, 5)

# Create PCA object
pca = PCA()

# Fit PCA to the data (training)
X_PCA = pca.fit_transform(X)


# Print explained variance ratio
print(pca.explained_variance_ratio_)

# Print feature loadings (optional)
print("Feature loadings:")
print(pca.components_)
